synapse/lib/types.py

```diff
@@ -613,6 +614,55 @@ def _normPyTuple(self, valu):
     def indx(self, norm):
         return s_common.buid(norm)
 
+class Hex(Type):
+    _opt_defs = (
+        ('width', 0),
+    )
+
+    def postTypeInit(self):
+        self._width = self.opts.get('width')
+        if self._width < 0:
+            # zero means no width check
+            raise s_exc.BadConfValu(name='width', valu=self._width,
+                                    mesg='Width must be > 0')
```

Comment:
Error message says that width must be > 0, but you could set it to 0 and pass this check.

Reply:
We have a problem of a "unknown length field" or  "known length field". width=0 means unknown length; whereas width > 0 means a known length.  A known length field can have its storage size precomputed.

Reply:
Ah, I would have thought that width=None would mean no known length and 0 would have been invalid